software/base/src/test/java/brooklyn/entity/basic/SoftwareProcessEntityTest.java

```diff
@@ -568,4 +706,85 @@ protected String getInstallLabelExtraSalt() {
             return (String)getEntity().getConfigRaw(ConfigKeys.newStringConfigKey("salt"), true).or((String)null);
         }
     }
+
+    public static class ReleaseLatchLocation extends SimulatedLocation {
+        private static final long serialVersionUID = 1L;
+        
+        private AtomicBoolean lock = new AtomicBoolean();
+        private boolean isBlocked;
+
+        public void unblock() {
+            synchronized (lock) {
+                lock.set(true);
+                lock.notifyAll();
+            }
+        }
+        @Override
+        public void release(MachineLocation machine) {
+            super.release(machine);
+            synchronized (lock) {
+                if (!lock.get()) {
+                    try {
+                        isBlocked = true;
+                        lock.wait();
+                        isBlocked = false;
+                    } catch (InterruptedException e) {
+                        throw Exceptions.propagate(e);
+                    }
+                }
+            }
+        }
+
+        public boolean isBlocked() {
+            return isBlocked;
+        }
+
+    }
+
+    public static class GenericSoftwareProcessTestDriver implements SoftwareProcessDriver {
```

Comment:
For the future (not this PR)...
Feels like this is useful for more than just here. For example, we already have:
* `DoNothingSoftwareProcessDriver`
* `SoftwareProcessEntityTest.SimulatedDriver`
* `SoftwareProcessSshDriverIntegrationTest.SimulatedDriver`
* `MockSshDriver` (which is a stub rather than mock)
* `RestMockSimpleEntity.MockSshDriver`